test/object-store/c_api/c_api.cpp

```diff
@@ -6024,40 +6024,46 @@ TEST_CASE("C API app: websocket provider", "[sync][app][c_api][baas]") {
 
     struct TestWebSocketObserverShim : sync::WebSocketObserver {
     public:
-        explicit TestWebSocketObserverShim(std::shared_ptr<sync::WebSocketObserver> observer)
+        explicit TestWebSocketObserverShim(realm_websocket_observer_t* observer)
             : m_observer(observer)
         {
         }
 
         void websocket_connected_handler(const std::string& protocol) override
         {
-            return m_observer->websocket_connected_handler(protocol);
+            REALM_ASSERT(m_observer);
+            realm_sync_socket_websocket_connected(m_observer, protocol.c_str());
         }
 
         void websocket_error_handler() override
         {
-            m_observer->websocket_error_handler();
+            REALM_ASSERT(m_observer);
+            realm_sync_socket_websocket_error(m_observer);
         }
 
         bool websocket_binary_message_received(util::Span<const char> data) override
         {
-            return m_observer->websocket_binary_message_received(data);
+            REALM_ASSERT(m_observer);
+            return realm_sync_socket_websocket_message(m_observer, data.data(), data.size());
         }
 
         bool websocket_closed_handler(bool was_clean, WebSocketError error, std::string_view msg) override
         {
-            return m_observer->websocket_closed_handler(was_clean, error, msg);
+            REALM_ASSERT(m_observer);
+            return realm_sync_socket_websocket_closed(m_observer, was_clean,
+                                                      static_cast<realm_web_socket_errno_e>(error), msg.data());
         }
 
     private:
-        std::shared_ptr<sync::WebSocketObserver> m_observer;
+        realm_websocket_observer_t* m_observer;
```

Comment:
I assume whoever creates a TestWebSocketObserverShim releases the realm_websocket_observer_t?

Reply:
The `realm_websocket_observer_t` will automatically be released by `~CAPIWebSocket()` and the `TestWebSocketObserverShim` will be deleted when the DefaultWebSocketImpl is destroyed.